sdk/storage/inc/blobs/protocol/blob_rest_client.hpp

```diff
@@ -1473,6 +1830,302 @@ namespace Azure { namespace Storage { namespace Blobs {
+            }
+            else if (std::strcmp(node.Name, "IncludeAPIs") == 0)
+            {
+              path.emplace_back(XmlTagName::k_IncludeAPIs);
+            }
+            else if (std::strcmp(node.Name, "RetentionPolicy") == 0)
+            {
+              path.emplace_back(XmlTagName::k_RetentionPolicy);
+            }
+            else
+            {
+              path.emplace_back(XmlTagName::k_Unknown);
+            }
+            if (path.size() == 1 && path[0] == XmlTagName::k_RetentionPolicy)
+            {
+              ret.RetentionPolicy = BlobRetentionPolicyFromXml(reader);
+              path.pop_back();
+            }
+          }
+          else if (node.Type == XmlNodeType::Text)
+          {
+            if (path.size() == 1 && path[0] == XmlTagName::k_Version)
+            {
+              ret.Version = node.Value;
+            }
+            else if (path.size() == 1 && path[0] == XmlTagName::k_Enabled)
+            {
+              ret.Enabled = std::strcmp(node.Value, "true") == 0;
+            }
+            else if (path.size() == 1 && path[0] == XmlTagName::k_IncludeAPIs)
+            {
+              ret.IncludeApis = std::strcmp(node.Value, "true") == 0;
+            }
+          }
+        }
+        return ret;
+      }
+
+      static BlobRetentionPolicy BlobRetentionPolicyFromXml(XmlReader& reader)
+      {
+        BlobRetentionPolicy ret;
+        enum class XmlTagName
+        {
+          k_Enabled,
+          k_Days,
+          k_Unknown,
+        };
+        std::vector<XmlTagName> path;
+        while (true)
+        {
+          auto node = reader.Read();
+          if (node.Type == XmlNodeType::End)
+          {
+            break;
+          }
+          else if (node.Type == XmlNodeType::EndTag)
+          {
+            if (path.size() > 0)
+            {
+              path.pop_back();
+            }
+            else
+            {
+              break;
+            }
+          }
+          else if (node.Type == XmlNodeType::StartTag)
+          {
+            if (std::strcmp(node.Name, "Enabled") == 0)
+            {
+              path.emplace_back(XmlTagName::k_Enabled);
+            }
+            else if (std::strcmp(node.Name, "Days") == 0)
+            {
+              path.emplace_back(XmlTagName::k_Days);
+            }
+            else
+            {
+              path.emplace_back(XmlTagName::k_Unknown);
+            }
+          }
+          else if (node.Type == XmlNodeType::Text)
+          {
+            if (path.size() == 1 && path[0] == XmlTagName::k_Enabled)
+            {
+              ret.Enabled = std::strcmp(node.Value, "true") == 0;
+            }
+            else if (path.size() == 1 && path[0] == XmlTagName::k_Days)
+            {
+              ret.Days = std::stoi(node.Value);
+            }
+          }
+        }
+        return ret;
+      }
+
+      static BlobStaticWebsite BlobStaticWebsiteFromXml(XmlReader& reader)
```

Comment:
I remember we had a conversation regarding if the name should contain `Blob` in it. `Azure::Storage::Blobs::BlobSomething` just feels redundant. I think we want to bring this up to Jeffrey to avoid getting challenged during official API review.

Reply:
Sure, we can certainly discuss about this with Jeffrey. But currently it aligns with .Net, so I guess we're unlikely to be challenged.